pkg/client/tools.go

```diff
@@ -123,11 +166,97 @@ func ImageImportIntoClusterMulti(ctx context.Context, runtime runtimes.Runtime,
 			l.Log().Errorf("failed to delete tools node '%s' (try to delete it manually): %v", toolsNode.Name, err)
 		}
 	}
+	return nil
+}
 
-	l.Log().Infoln("Successfully imported image(s)")
+func importWithStream(ctx context.Context, runtime runtimes.Runtime, imagesFromRuntime []string, cluster *k3d.Cluster, imagesFromTar []string) error {
+	if len(imagesFromRuntime) > 0 {
+		l.Log().Infof("Loading %d image(s) from runtime into nodes...", len(imagesFromRuntime))
+		// open a stream to all given images
+		stream, err := runtime.GetImageStream(ctx, imagesFromRuntime)
+		loadImageFromStream(ctx, runtime, stream, cluster)
```

Comment:
Don't we need errors returned here?

Reply:
So if this is failing, we will print a log line, but the command will still return successfully, right? :thinking:

Reply:
yep, sorry i didn't finish this yesterday; should be good now IMO (i migrated the waitgroup to an errgroup so it returns errors)